pkg/modprovider/planstore.go

```diff
@@ -156,6 +188,70 @@ func (e unknownAddressError) Error() string {
 	return fmt.Sprintf("unknown address: %q", e.addr)
 }
 
+// IsResourceDeleted returns true if the resource is deleted (not in the state).
+//
+// There are a couple different delete cases that this function handles:
+//  1. This is a `pulumi dn` operation and the entire stack is being deleted
+//     In this case there will be no `plan` because the `moduleStateHandler.Delete` method does not run plan.
+//  2. This is a `pulumi up` operation and this resource is being deleted or replaced.
+//     - 2a.If DeleteBeforeReplace=false (TF default) then `Delete` is being run in the same provider process
+//     and we have access to shared info (i.e. plan/state).
+//     We can tell this case apart from 1 because there will be a plan (`up` runs Construct which runs Plan).
+//     - 2b. If DeleteBeforeReplace=true then `Delete` is being run in a different provider process
+//     and we do not have access to shared info (i.e. plan/state). We don't have any info to tell
+//     if the delete succeeded or not for this resource
+//     - 2c. If this resources is being deleted and not replaced (i.e. resource is removed) then the `Delete`
+//     will happen in a separate provider process and we will not have access to shared info (state/plan)
+//
+// NOTE: This should only be called from within the `Delete` method of the child resource
+//
+// TODO[pulumi/pulumi-terraform-module#265] determine if the delete for this
```

Comment:
Nit: perhaps this logic is not really part of the planStore.. separate file? I can't find a problem with it per se but it's really complicated. We have design problems..